ivadomed/loader/files_dataset.py

```diff
@@ -0,0 +1,239 @@
+                logger.warning(f"Fewer input files found than expected for subject specification "
+                               f"{a_subject_image_ground_truth_pair}. Expected {n_expected_input} "
+                               f"but found {len(list_subject_specific_images)} from {list_subject_specific_images}")
+                if self.drop_missing:
+                    continue
+
+            if len(list_subject_specific_gt) < n_expected_gt:
+                logger.warning(f"Fewer ground truth files found than expected, for subject specification "
+                               f"{a_subject_image_ground_truth_pair}. Expected {n_expected_gt} "
+                               f"but found {len(list_subject_specific_gt)} from {list_subject_specific_gt}")
+                if self.drop_missing:
+                    continue
+
+            for gt_index, path_file in enumerate(list_subject_specific_gt):
+                list_subject_specific_gt[gt_index] = ensure_absolute_path(
+                    path_file, path_data
+                )
+                if not list_subject_specific_images[gt_index]:
+                    skip_subject_flag = True
+
+            # Exclude current subject
+            if skip_subject_flag:
+                continue
+
+            # Generate simple meta data #todo: should load json if present.
+            metadata: List[dict] = [{}]* len(list_subject_specific_images) # "data_specificiation_type": "file_dataset"
+
+            list_image_ground_truth_pairs_filtered.append((list_subject_specific_images, list_subject_specific_gt))
+
+            # At this point, established ALL subject's related image and ground truth file exists.
+            filename_pairs.append(
+                (
+                    list_subject_specific_images,
+                    list_subject_specific_gt,
+                    None,  # No ROI for this dataset, String?
+                    metadata,  # No metadata for this dataset, Dict?
+                )
+            )
+
+        loader_json[DataloaderKW.IMAGE_GROUND_TRUTH] = list_image_ground_truth_pairs_filtered
+
+        return filename_pairs
+
+    def preview(self, verbose: bool = False):
+        """
+        Preview the FINALIZED FileDataSet included that has undergone validation and is ready to be used for training.
+        Args:
+            verbose: whether to print out the actual data path
+        """
+        logger.info(f"\t\tFileDataset object from {self.path_data}, {len(self.filename_pairs)} pairs of data files")
+
+        if verbose:
+            for pair_index, a_pair in enumerate(self.filename_pairs):
+                logger.info(f"\t\t\tImage Pair {pair_index}, Subject Image(s):")
+                for a_image in a_pair[0]:
+                    logger.info(f"\t\t\t\t{a_image}")
+                logger.info(f"\t\t\tImage Pair {pair_index}, Ground Truth Image(s):")
+                for a_gt in a_pair[1]:
+                    logger.info(f"\t\t\t\t{a_gt}")
```

Comment:
Since this was unaligned in terms of whitespace and a little unintuitive to me so I refactored it to:
```
    def preview(self, verbose: bool = False):
        """
        Preview the FINALIZED FileDataSet included that has undergone validation and is ready to be used for training.
        Args:
            verbose: whether to print out the actual data path
        """
        logger.info(f"FileDataset object from {self.path_data}, {len(self.filename_pairs)} pairs of data files")
        
        if verbose:
            for pair_index, (imgs, gts, _, _) in enumerate(self.filename_pairs):
                logger.info(f"Image Pair {pair_index}, Subject Image(s):")                
                for img in imgs:
                    logger.info(f"\t{img}")
                
                logger.info(f"Image Pair {pair_index}, Ground Truth Image(s):")
                for gt in gts:
                    logger.info(f"\t{gt}")
```
In a lot of places I feel we could use something like 
`for pair_index, (imgs, gts, _, _) in enumerate(self.filename_pairs):` which makes everything more readable 🤷‍♂️